docs/develop/dialog-api-in-office-add-ins.md

```diff
@@ -293,11 +330,48 @@ function onRegisterMessageComplete(asyncResult) {
+> In some situations, the `messageChild` API, which is a part of the [DialogApi 1.2 requirement set](../reference/requirement-sets/dialog-api-requirement-sets.md), may not be supported. Some alternative ways for parent-to-dialog-box messaging are described in [Alternative ways of passing messages to a dialog box from its host page](parent-to-dialog.md).
 
 > [!IMPORTANT]
 > The [DialogApi 1.2 requirement set](../reference/requirement-sets/dialog-api-requirement-sets.md) cannot be specified in the `<Requirements>` section of an add-in manifest. You will have to check for support for DialogApi 1.2 at runtime using the [isSetSupported](specify-office-hosts-and-api-requirements.md#use-runtime-checks-in-your-javascript-code) method. Support for manifest requirements is under development.
 
+### Cross-domain messaging to the dialog runtime
+
+The dialog may be navigated away from the add-in's domain or the parent JavaScript runtime (either in a task pane or a UI-less runtime that hosts a function file) may be navigated away from the add-in's domain after the dialog is opened. If either of these things has happened, then a call of `messageChild` will fail unless your code specifies the domain of the dialog runtime. You do this by adding a [DialogMessageOptions](/javascript/api/office/office.dialogmessageoptions) parameter to the call of `messageChild`. This object has a `targetOrigin` property that specifies the domain to which the message should be sent. If the parameter is not used, Office assumes that the target is the same domain that the parent runtime is currently hosting. 
+
+> [!NOTE]
+> Using `messageChild` to send a cross-domain message requires the [Dialog Origin 1.1 requirement set](../reference/requirement-sets/dialog-origin-requirement-sets.md).
+
+The following is an example of using `messageChild` to send a cross-domain message.
+
+```js
+dialog.messageChild(messageToDialog, { targetOrigin: "https://target.domain.com" });
+```
+
+If the message doesn't include sensitive data, you can set the `targetOrigin` to "\*" which allows it to be *sent* to any domain. The following is an example.
+
+```js
+dialog.messageChild(messageToDialog, { targetOrigin: "*" });
+```
+
+The JavaScript runtime that is hosting the dialog cannot access the `<AppDomains>` section of the manifest to ensure that the domain *from which the message comes* is trusted, so an alternative has been provided. The object that is passed to the `DialogParentMessageReceived` handler contains the domain that is currently hosted in the dialog as its `origin` property. Your code should use the handler to verify that the message is coming from a trusted domain. The following is an example.
```

Comment:
Maybe adjust to something along the lines of 
"Because the JavaScript runtime hosting the dialog can't access the `<AppDomains>` section of the manifest and therefore can't determine if a message's domain is trusted, you must use the `DialogParentMessageReceived` handler to verify trusted messages. The `DialogParentMessageReceieved` handler contains the domain that is currently hosted in the dialog as its `origin` property. The following is an example:"